2022/Day7/Solve.py

```diff
@@ -0,0 +1,83 @@
+class Day_Six(object):
+  def day_six(self):
+    computer = self.parse("PuzzleInput.txt")
+    computer.calculate_size()
+    print("The answer to part one is", computer.sum_all_subdirectories_under_limit(100000))
+
+  def parse(self, puzzle_input):
+    current_directory = []
+    computer = Directory("dir /")
+    with open(puzzle_input, "r") as f:
+      for line in f:
+        if "cd /" in line:
+          continue
+        if ".." in line:
+          current_directory.pop()
+          continue
+        elif "$ cd" in line:
+          current_directory.append("dir " + line.split()[2])
```

Comment:
This is a really interesting choice.  Why not strip `dir` from the name of the directory files?

Reply:
In the AoC thread with Elliott, your grep showed that files AND directories could have the same name. I wanted a way to clearly create a distinction so that it wouldn't try and put files within files accidentally.

Reply:
Hmm - you already have a Directory class; you could be utilizing a Directory instance here as part of your current directories contents.

Reply:
Hmmm, thinking about this further; you are maintaining a master table of files and directories within `computer`; and maintaining `current_directory` as a reference point to where you are in your filesystem. However; you have a perfectly good hierarchy defined with the Directory class, which is aware of obviously itself, and the paths underneath it. The only thing it might be missing is knowing who it's parent is, and you have a node structure that would be traversable.

Reply:
I'm not sure what you mean, @elliottcarlson

Reply:
Actually - disregard; I made a wrong assumption of the code. Pulled it down to execute and check things out.

Reply:
Actively traversing the node tree would be even cleaner, for sure.

@glopez0909 the idea would be instead of maintaining "current_directory" as a map, you'd maintain just a pointer to the "current directory":

- directory.add would just take a new file
- you'd add a directory.get(name) to get the thing you're cding into
- you'd add a directory.parent() to go up a level, and your code would look something like:

```
current_directory  = Directory("/")
on cd ..:
  current_directory = current_directory.parent()
on cd name:
  current_directory = current_directory.get(name)
```

It would require you add a little bit of extra management in your objects, which I don't think Elliot wanted to sidetrack you with here.  I think it might be worth looking into at a future time, when you're more comfortable with tree structures.